app/src/androidTest/java/com/android/sample/endToend/EndToEndTest1.kt

```diff
@@ -0,0 +1,144 @@
+package com.android.sample.endToend
+
+import android.content.Context
+import android.content.Intent
+import androidx.compose.ui.test.assertHasClickAction
+import androidx.compose.ui.test.assertIsDisplayed
+import androidx.compose.ui.test.assertTextEquals
+import androidx.compose.ui.test.junit4.createComposeRule
+import androidx.compose.ui.test.onNodeWithTag
+import androidx.compose.ui.test.onNodeWithText
+import androidx.compose.ui.test.performClick
+import androidx.core.content.ContextCompat.startActivity
+import androidx.test.core.app.ApplicationProvider
+import com.android.sample.MainActivity
+import com.android.sample.ui.navigation.NavigationActions
+import com.android.sample.ui.navigation.Route
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+class EndToEndTest1 {
+  private lateinit var navigationActions: NavigationActions
+  @get:Rule val composeTestRule = createComposeRule()
+
+  @Before
+  fun setUp() {
+    val intent = Intent(ApplicationProvider.getApplicationContext(), MainActivity::class.java)
+    intent.putExtra("START_DESTINATION", Route.MAIN)
+    val context = ApplicationProvider.getApplicationContext<Context>()
+    intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+    context.startActivity(intent)
+  }
+
+  @Test
+  fun testEndToEnd() {
+    // Test begins at the main screen
+    // check that everything is displayed
+    mainScreenIsWellDisplayed()
+    // Go to the achievement screen
+    composeTestRule.onNodeWithTag("Achievements").performClick()
+    achievementScreenIsWellDisplayed()
+    // go back to the main screen
+    composeTestRule.onNodeWithTag("Main").performClick()
```

Comment:
Check the new implementation of the main screen as some the test tags might have changed